### PR TITLE
Alire.Os_Lib.Subprocess refactoring

### PR DESCRIPTION
**Pull request by Fabien-Chouteau**
The main goal of this patch is to move all the sub-process APIs to use
arguments in the form of a vector of strings rather than a single string
that is later sliced into different arguments. The reason for this
change is to better handle white-spaces in arguments, in particular
file-system paths.

The patch also removes unused or duplicate sub-programs in
Alire.Os_Lib.Subprocess and Alr.Os_Lib.

Because of the change of actions.command format in the manifests, this
patch requires an update of some crates in the index.

Part of #253 and #252.

### CODE REVIEW

**Comment by Fabien-Chouteau** [comment]
Unfortunately I think that the testsuite does not cover all the changes made in this patch, in particular `hg` and `svn` VCSs .

Before merging it would be better to add at least a test on the `action.command` of the manifest.

**Comment by mosteo** [comment]
> Unfortunately I think that the testsuite does not cover all the changes made in this patch, in particular `hg` and `svn` VCSs .

We aren't using `hg` or `svn` anywhere anymore, I think...

File: src/alire/alire-origins-deployers-apt.adb [hunk 1]
```diff
@@ -42,19 +43,21 @@ package body Alire.Origins.Deployers.APT is
...
-                            This.Base.Package_Name (Platform.Distribution));
-
-      if Exit_Code /= 0 then
-         return Outcome_Failure ("apt-get install exited with code:"
-                                 & Exit_Code'Img);
-      end if;
+      Subprocess.Checked_Spawn
+        ("sudo", Empty_Vector &
+           "apt-get" &
+           "instal" &
```
Comment (mosteo) [inline src/alire/alire-origins-deployers-apt.adb]: install typo

**Comment by mosteo** [comment]
> Before merging it would be better to add at least a test on the `action.command` of the manifest.

Waiting for this before merging, then.

File: src/alire/alire-origins-deployers-apt.adb (hunk 1 above)
Comment (Fabien-Chouteau) [reply src/alire/alire-origins-deployers-apt.adb]: Good  catch!

**Comment by Fabien-Chouteau** [comment]
I changed the `actions.command` definition to only accept array of string(s). In the previous version single string was accepted but I realized that it could be error prone to accept `command = "make something"`

**Comment by mosteo** [comment]
Great, it's surely more robust. Also, there are no ambiguities accepted in any other place in the index format either (in the sense of accepting a single value where an array is expected).

**Comment by Fabien-Chouteau** [comment]
> We aren't using `hg` or `svn` anywhere anymore, I think...

Should we remove them?

**Comment by mosteo** [comment]
> Should we remove them?

I'd just leave them there, if they're not giving you extra work. Otherwise, I don't foresee a clear need.

**Comment by Fabien-Chouteau** [comment]
> I'd just leave them there, if they're not giving you extra work. Otherwise, I don't foresee a clear need for them.

The problem is that this is more or less dead code.

**Comment by mosteo** [comment]
Out with it, then.

**Comment by Fabien-Chouteau** [comment]
The code is not that big and it will be possible to bring it back if need be.

Let's remove that in a separate PR though.

**Comment by mosteo** [comment]
OK! I will merge this one shortly.

**Comment by mosteo** [comment]
To be merged with https://github.com/alire-project/alire-index/pull/72

**Pull request merged**